src/main/java/org/fenixedu/academic/domain/Degree.java

```diff
@@ -615,6 +617,20 @@ final public MultiLanguageString getNameI18N() {
     final public MultiLanguageString getNameI18N(ExecutionYear executionYear) {
         return getNameFor(executionYear);
     }
+    
+    public LocalizedString getPresentationNameI18N() {
+        return getPresentationNameI18N(ExecutionYear.readCurrentExecutionYear());
+    }
+    
+    public LocalizedString getPresentationNameI18N(final ExecutionYear executionYear) {
```

Comment:
This is mostly a reconstruction of a localized string from broken bits of what should be a localized string since the beginning. Make the getPresentationNameI18N the source method from all getPresentationName* methods, and implement it like so:

```java
    public LocalizedString getPresentationNameI18N(final ExecutionYear executionYear) {
        LocalizedString in = BundleUtil.getLocalizedString(Bundle.APPLICATION, "label.in");
        LocalizedString degreeType = getDegreeType().getName();
        Builder builder = degreeType.builder();
        degreeType.getLocales().forEach(l -> builder.append(in.getContent(l), " "));
        builder.append(getNameFor(executionYear).toLocalizedString(), " ");
        return builder.build();
    }
```

All other just invoke this, with variations of ```ExecutionYear.readCurrentExecutionYear()``` as a parameter or ```.getContent()``` invoked after.